language/src/regent/optimize_index_launches.t

```diff
@@ -472,6 +474,9 @@ local function analyze_noninterference_self(
     if analyze_index_noninterference_self(index, cx, loop_vars)
     then
       return true
+    else
+      local needs_dynamic_check = true
```

Comment:
What's the reason for the local variable here?

Reply:
I had hoped this would improve readability when we return `false, true` on the next line, but I'm not confident that this is the case.